### PR TITLE
webui: start using payload from the bots repository

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

File: ui/webui/test/machine_install.py [hunk 1]
```diff
@@ -62,18 +62,44 @@ def _create_disk_image(self, size, image_path=None, quiet=False):
         self._execute(f"qemu-img create -f qcow2 {quiet} {image_path} {size}G")
         return image_path
 
+    def _serve_updates_img(self):
+        http_port = self._get_free_port()
+        self.http_server = subprocess.Popen(["python3", "-m", "http.server", "-d", ANACONDA_ROOT_DIR, str(http_port)])
+        self.http_server.communicate()
```
Comment (martinpitt) [inline ui/webui/test/machine_install.py:68]: This is the problem with the forever-hanging test. communicate() waits until the spawned process exits, which will never happen until you kill it. Instead, use `m.wait_for_cockpit_running('127.0.0.1', http_port)` until it is up, let it run in the background, and set up a cleanup handler which does `.terminate()` and `.wait()`.

File: ui/webui/test/machine_install.py [hunk 2]
```diff
@@ -62,18 +63,52 @@ def _create_disk_image(self, size, image_path=None, quiet=False):
         self._execute(f"qemu-img create -f qcow2 {quiet} {image_path} {size}G")
         return image_path
 
+    def _wait_http_server_running(self, port):
```
Comment (martinpitt) [inline ui/webui/test/machine_install.py:66]: What is wrong with the existing `wait_for_cockpit_running('127.0.0.1', http_port)` [code](https://github.com/cockpit-project/bots/blob/000d0e34b42a978ad13b6ab55219fbff8296eeff/machine/machine_core/machine.py#L320) ? You can't access that API from anaconda, due to the different Machine class?

**Review by martinpitt — CHANGES_REQUESTED** [review]
Nice, thanks!

File: ui/webui/test/machine_install.py (hunk 2 above)
Comment (martinpitt) [reply ui/webui/test/machine_install.py:66]: Yeah, the name is confusing, it should really be called `wait_for_http_port`..

File: ui/webui/test/machine_install.py [hunk 3]
```diff
@@ -62,18 +63,52 @@ def _create_disk_image(self, size, image_path=None, quiet=False):
...
+
+        return http_port
+
+    def _serve_payload(self):
+        payload_cached_path = os.path.realpath(os.path.join(BOTS_DIR, "./images/fedora-rawhide-anaconda-payload"))
+        payload_cached_dir = os.path.dirname(payload_cached_path)
+        payload_cached_name = os.path.basename(payload_cached_path)
+
+        http_payload_port = self._get_free_port()
+        self.http_payload_server = subprocess.Popen(["python3", "-m", "http.server", "-d", payload_cached_dir, str(http_payload_port)])
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply ui/webui/test/machine_install.py:88]: Fixed.

File: ui/webui/test/machine_install.py (hunk 2 above)
Comment (martinpitt) [reply ui/webui/test/machine_install.py:66]: Argh, my bad :see_no_evil:

**Review by martinpitt — APPROVED** [review]
Cheers!

**Pull request merged**